Source/WebCore/css/html.css

```diff
@@ -794,6 +794,10 @@ input::placeholder {
     line-height: initial !important;
 }
 
+input:disabled, textarea:disabled {
+    color: color-mix(in hsl, CanvasText 66%, Canvas);
```

Comment:
This new color is not dependent on the background color of the form element. So this means that in some cases when the background color is set, the old code would select a color that was visible and this new approach will allow the text to be invisible.

Perhaps that is not an important problem to solve?

Did you look at what the old and new approach do in various contexts aside from the ones where the color is set explicitly? I suppose if other web browsers don’t try to tackle this problem, we can just drop it. We are deciding that it’s up to the website designer to customize both the text and the background color.

Reply:
> Did you look at what the old and new approach do in various contexts aside from the ones where the color is set explicitly? I suppose if other web browsers don’t try to tackle this problem, we can just drop it. We are deciding that it’s up to the website designer to customize both the text and the background color.

Other browsers just let the custom color also override the disabled color, there isn't any adjustment done.

Reply:
This is the result in between different browsers and the patch. The situation is not all bad or all good. It depends on the case. So the previous algorithm was not also always a way to get a perfect contrasted rendering.

![Screenshot 2023-02-14 at 7 37 56](https://user-images.githubusercontent.com/505230/218590937-c3baf07e-e52f-40a2-b782-8563789cf8c9.png)

As @nt1m said the benefit here is that before designers were not able to control the color at all for the `disabled` mode, but now they can override the color to adjust to their choice colors matching their website design.

Reply:
Yes, but that undeniable benefit — website authors can override the color — is mostly orthogonal to the thing I was questioning, the change to which color we choose when they **do not** override the color.

If our old choice of color was really good, we could have fixed this in a way that preserved our old algorithm while supporting overriding. The new color choice algorithm has to be judged on its merits.

I accept the screen shot as evidence that the new algorithm is better than the old. I am surprised that it somehow responds to the background. I didn’t realize how powerful `Canvas` and `CanvasText` were.